pre-commit

```diff
@@ -46,4 +46,6 @@ error() {
 
 cargo clippy --tests -- -D warnings --D clippy::pedantic || error "Clippy errors"
 
+./install_and_start_redis
```

Comment:
This only works for the pre-commit hook.

I think that it is reasonable to assume that people already have a redis-server installed locally if they are doing work on this project.  But the tests should be responsible for starting their own server instance as necessary.

If you don't want to assume anything, you can add a build.rs file and have that install redis-server into the target/ directory.

Reply:
Agree that build.rs would be a better approach. For now, all developers are implementing their modules so dont really need to "ensure" a running redis server. Will move it to build.rs